pandas/_libs/reduction.pyx

```diff
@@ -493,7 +493,7 @@ def apply_frame_axis0(object frame, object f, object names,
             # Need to infer if low level index slider will cause segfaults
             require_slow_apply = i == 0 and piece is chunk
             try:
-                if piece.index is not chunk.index:
+                if not piece.index.equals(chunk.index):
```

Comment:
Is this something we should perform benchmarking before merging?

Reply:
this is cheap